### PR TITLE
cargo-espflash: rename to espflash

### LINKED ISSUE #285240: Similar derivations: cargo-espflash and espflash
## Issue description

Recently (https://github.com/NixOS/nixpkgs/pull/275490) the new package `espflash` was added to nixpkgs.
But the outputs are extremely similar to the existing package `cargo-espflash`, except that it doesn't contain the cargo wrapper executable:

```
$ nix build .#espflash
$ find ./result/
./result/
./result/bin
./result/bin/espflash
$ nix build .#cargo-espflash
$ find ./result/
./result/
./result/bin
./result/bin/cargo-espflash
./result/bin/espflash
```

and the nixpkgs source files for [espflash](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/embedded/espflash/default.nix) and the existing [cargo-espflash](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/tools/rust/cargo-espflash/default.nix) are very similar as well.

Does it really make sense to have them as completly separate derivations? It confused me for a while trying to find out if there were any non-obvious differences.

@matthiasbeyer @newAM

## Technical details

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

### PR DESCRIPTION
**Pull request by newAM**
## Description of changes

This binary already exists within the cargo-espflash package.

Closes #285240 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -31287,10 +31287,6 @@ with pkgs;
 
   espeakup = callPackage ../applications/accessibility/espeakup { };
 
-  espflash = callPackage ../development/embedded/espflash {
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix:31290]: We might want to add an entry to `pkgs/top-level/aliases.nix` instead.

File: pkgs/development/tools/rust/cargo-espflash/default.nix [hunk 2]
```diff
@@ -43,6 +43,8 @@ rustPlatform.buildRustPackage rec {
   meta = with lib; {
     description = "Serial flasher utility for Espressif SoCs and modules based on esptool.py";
     homepage = "https://github.com/esp-rs/cargo-espflash";
```
Comment (dotlambda) [inline pkgs/development/tools/rust/cargo-espflash/default.nix:45]:
```suggestion
    homepage = "https://github.com/esp-rs/espflash";
```

**Review by dotlambda — COMMENTED** [review]
Maybe we should actually rename the package?

**Comment by newAM** [comment]
I don't have any strong opinion on renaming.  Added the alias and fixed the 404 link.

**Comment by veprbl** [comment]
@GrahamcOfBorg build cargo-espflash

**Comment by dotlambda** [comment]
We are the only ones who call it espflash-cargo: https://repology.org/projects/?search=espflash
Let's rename.

**Comment by newAM** [comment]
> We are the only ones who call it cargo-espflash

That's a good enough reason for me.  Done.

**Comment by veprbl** [comment]
```
error: hash mismatch in fixed-output derivation '/nix/store/bvjhi958f2bf1grpcrhrlzhhhhq7kb7s-espflash-2.1.0-vendor.tar.gz.drv':
         specified: sha256-FpBc92a2JQHRLe5S6yh3l0FpRI8LpkGGEma/4v5X4xs=
            got:    sha256-Xj5FVTssC3e+mMhDHmKqV6lUQgaIv3aVc1yewbQSy9E=
error: 1 dependencies of derivation '/nix/store/mbqqvhwlkqh6wr1ka467zmc98jy0znyb-espflash-2.1.0.drv' failed to build
```

**Comment by newAM** [comment]
My bad!  I didn't know changing the pname would change the hash.

**Comment by newAM** [comment]
Result of `nixpkgs-review pr 286386` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>espflash</li>
  </ul>
</details>

**Comment by veprbl** [comment]
> My bad! I didn't know changing the pname would change the hash.

cc #85703

**Comment by veprbl** [comment]
@GrahamcOfBorg build espflash

**Pull request merged**